src/utils/asyncErrorHandler.ts

```diff
@@ -0,0 +1,13 @@
+import { Request, Response, NextFunction } from 'express';
```

Comment:
What is the benefits of this error handler? Could you just wrap code inside each controller in try/catch block and use next() callback inside catch if error occurs

Reply:
Middleware is responsible for error handling. Controller just have to pass control to middleware if error occurs.
I agree that in your approach it's not needed to wrap each controller in try/catch but anyway you use asyncErrorHandler everywhere.